presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -1008,6 +1009,18 @@ private Expression extractJoinPredicate(JoinNode joinNode)
             ImmutableList.Builder<Expression> builder = ImmutableList.builder();
             for (JoinNode.EquiJoinClause equiJoinClause : joinNode.getCriteria()) {
                 builder.add(equiJoinClause.toExpression());
+                switch (joinNode.getType()) {
```

Comment:
If the goal is to push down an IS NOT NULL clause to the left/right side, this is not the right place to do it. `extractJoinPredicate`'s job is to fetch the join condition as it appears in the join node. The clauses should be added inside `processInnerJoin` and `processLimitedOuterJoin` as additional expressions in the corresponding branches of `InnerJoinPushDownResult` and `OuterJoinPushDownResult`.